doc/advanced_configuration.rst

```diff
@@ -397,6 +398,50 @@ The default behavior is ``sphinx_gallery_thumbnail_number = 1``. See
 :ref:`sphx_glr_auto_examples_plot_choose_thumbnail.py` for an example
 of this functionality.
 
+.. _binder_links:
+
+Generate Binder links for gallery notebooks
+===========================================
+
+Sphinx-Gallery automatically generates Jupyter notebooks for any
+examples built with the gallery. `Binder <http://mybinder.org>`_ makes it
+possible to create interactive GitHub repositories connect to cloud resources.
+If you host your documentation on a GitHub repository, it is possible to
+auto-generate a Binder link for each notebook. Clicking this link will
+take users to a live version of the Jupyter notebook where they may
+run the code interactively. For more information see the `Binder documentation
+<http://docs.mybinder.org>`_.
+
+.. warning::
+
+   Binder is still beta technology, so there may be instability in the
+   experience of users who click Binder links.
+
+In order to enable Binder links with Sphinx-Gallery, you must specify
+a few pieces of information in ``conf.py``. These are given as a nested
+dictionary following the pattern below::
+
+    sphinx_gallery_conf = {
+      ...
+      'binder': {
+         'org': '<github_org>',
+         'repo': '<github_repo>',
+         'url': '<binder_url>',  # URL serving binders (e.g. mybinder.org)
+         'branch': '<repo_branch>',  # Can also be a tag or commit hash
+         'dependencies': '<list_of_paths_to_dependency_files>'
```

Comment:
How does binder deal with dependencies? Does it take multiple dependecies files? or just one `requirements.txt` or `requirements.yml`? Is it name specific? I'm thinking we want to use a file `doc/binder_requirements.txt` to set our environment then the setup copies the file renaming it.

Reply:
The code in this PR seems to require that it ends with either `requirements.txt` or `environment.yml`.

The error message seems to say that it can be only `requirements.txt` or `environment.yml`.
```
ValueError: Must provide requirements path to at least one of ['requirements.txt', 'environment.yml']
```

Reply:
Indeed it requires at least one. But what if my list is:
```python
'dependecies' : ['requirements.txt', 'foo.txt', 'bar.yml']
```
@choldgraf does that hold a meaning for binder? Why do you need to allow for a list of dependencies?

Reply:
good catch @lesteve , that was there because I opened this PR back in May before we built extra functionality into Binder :-)

@Titan-C the reason for allowing a list of dependencies is in case the author of the repo needs more than one dependency file to run their docs. E.g., in Binder you can include and apt.txt file that will apt-install things, or a `postBuild` file that will run shell commands after building the environment.

What if we required at least one of `requirements.txt and environment.yml` and allowed for any other files on top of that. If we're assuming people are running python, one of these seems reasonable to require.

Reply:
oh - and yes it is filename-specific. If _has_ to be called `requirements.txt` or `environment.yml`. To get around clashes with other files of hte same name, it's also possible to put files into a folder called `binder` and binder will find those too. So if there's a project that has `requirements.txt` and `doc-requirements.txt` in the same folder, they could use a `binder` folder

Reply:
I think we need to update the code was updated to check that the filename has to be either requirement.txt or environment.yml. Adding a test would be great and while I am at it, the error message wording is a bit weird too (maybe a missing word somewhere).

Reply:
This still needs to be addressed in the documentation. Telling which are the required files, I would also like to put a link to the binder docs where it say what the dependency files do.